### PR TITLE
7298 Resolves unhandled error when loading index issue details page

### PR DESCRIPTION
**Pull request by johnclary**
Fixes [#7298](https://github.com/cityofaustin/atd-data-tech/issues/7298).

I can't re-produce this exact behavior locally or in a deploy preview, but I was able to force the same error by manually setting issue to `undefined` and by setting an invalid project ID in the URL. 

In any case, it is fixed.

### CODE REVIEW

**Comment by johnclary** [comment]
@frankhereford thanks for review. No rush!

File: components/shared/indexIssueDetails/IndexIssueDetails.js [hunk 1]
```diff
@@ -135,32 +136,48 @@ function InfoRow(props) {
...
+    );
+  }
+
+  const issue = issues.filter((issue) => issue.number === issue_number)[0];
+
+  if (!issue) {
+    // this would suggest the user provided an invalid issue ID in the URL
+    return (
+      <Alert variant="danger">
+        No project data found for project ID #{issue.number}
```
Comment (frankhereford) [inline components/shared/indexIssueDetails/IndexIssueDetails.js:167]: If the issue is `false`, my take is that you can count on not being able to access the number property of that object, which will throw an error -- very similar to the behavior you described in cityofaustin/atd-data-tech/issues/7298, but emanating from the new error message. 

image

https://deploy-preview-28--atd-product.netlify.app/projects/31337 shows the same error in the deploy preview for this PR.

**Review by frankhereford — CHANGES_REQUESTED** [review]
@johnclary - I left a comment where I think the root cause of the original issue is being invoked again in the changes of this PR. 

I branched off your branch and pushed a commit with a solution which may be workable. You can find it [here](https://github.com/cityofaustin/atd-product/commit/0370e42e791495c3a0dfe1bd262245101b418b16).

It's pulling the requested issue number to show the user from the router object instead of the empty / undefined issue object. 

I'm not able to recreate this issue either reliably, but I do get it on the production deployment if I enable my cache and reload it a few times. I think it's from Socrata dropping the TCP connection before returning any data, and that empty response getting cached -- but I don't know if I have evidence to support that. If I block my browser from making connections to the Socrata endpoint, the app throws an error, but it's an error to the user and not a JS error. 

If you want, I can compose a quick script to accept TCP connections, spin for a second or two, and then close the connection without sending anything down the pipe. That may be a reliable way to reproduce the problem, but it also may be overkill.

File: components/shared/indexIssueDetails/IndexIssueDetails.js (hunk 1 above)
Comment (johnclary) [reply components/shared/indexIssueDetails/IndexIssueDetails.js:167]: you're completely right—this was a boneheaded mistake on my part 🙏

Comment (johnclary) [reply components/shared/indexIssueDetails/IndexIssueDetails.js]: thanks @frankhereford. i just patched a fix by changing the reference to the `issue_number` prop. This value is pulled from the router query in the parent component.

Comment (johnclary) [reply components/shared/indexIssueDetails/IndexIssueDetails.js]: Screen Shot 2021-10-01 at 1 18 08 PM

**Comment by johnclary** [comment]
@frankhereford you mind giving this one more quick look for me? :)

**Comment by frankhereford** [comment]
Works like a charm!

**Review by frankhereford — APPROVED** [review]
Works like a charm!

**Pull request merged**